src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILInterpreter.cs

```diff
@@ -2442,5 +2445,163 @@ private void InterpretLoadElement(int token)
                     throw new NotImplementedException();
             }
         }
+
+        // Holds information about a method call
+        private struct MethodCallInfo
+        {
+            public int ArgCount;
+            public IntPtr MethodAddress;
+            public IntPtr UnboxingStubAddress;
+            public CallConverter.CallingConvention CallingConvention;
+            public TypeLoaderEnvironment.MethodAddressType MethodAddressType;
+            public LocalVariableType[] LocalVariableTypes;
+            public TypeDesc ReturnType;
+            public StackItem ReturnValue;
+        }
+
+        private void InterpretCallDelegate(ref MethodCallInfo callInfo, ref LocalVariableSet localVariableSet)
+        {
+            for (int i = callInfo.ArgCount; i > 0; i--)
+            {
+                StackItem stackItem = PopWithValidation();
+                switch (stackItem.Kind)
+                {
+                    case StackValueKind.Int32:
+                        localVariableSet.SetVar<int>(i, stackItem.AsInt32());
+                        break;
+                    case StackValueKind.Int64:
+                        localVariableSet.SetVar<long>(i, stackItem.AsInt64());
+                        break;
+                    case StackValueKind.NativeInt:
+                        localVariableSet.SetVar<IntPtr>(i, stackItem.AsNativeInt());
+                        break;
+                    case StackValueKind.Float:
+                        localVariableSet.SetVar<double>(i, stackItem.AsDouble());
+                        break;
+                    case StackValueKind.ByRef:
+                        // TODO: Add support for ByRef to StackItem
+                        throw new NotImplementedException();
+                    case StackValueKind.ObjRef:
+                        localVariableSet.SetVar<object>(i, stackItem.AsObjectRef());
+                        break;
+                    case StackValueKind.ValueType:
+                        localVariableSet.SetVar<ValueType>(i, stackItem.AsValueType());
```

Comment:
Handling of ValueTypes doesn't look right - I think it would be better to make those throw for now.

(This should eventually use `GetAddressOfVarData` and copy the bytes of the valuetype to that location.)

Reply:
Can you please explain further? Why isn't it enough to just set the value type directly as it's done here?

Reply:
`System.ValueType` is a reference type. This line does the same thing as the `localVariableSet.SetVar<object>` above - it will use the reference to the boxed valuetype as the argument, but the called method expects the value type to be passed by value.

Reply:
Aaah was thinking along those lines. Thanks for confirming